src/function/scalar/string/like.cpp

```diff
@@ -56,11 +93,23 @@ bool like_operator(const char *s, const char *pattern, const char *escape) {
 		} else {
 			return false;
 		}
+		is_escaped = false;
 	}
 	if (*p == '%' && *(p + 1) == 0) {
 		return true;
 	}
-	return *t == 0 && *p == 0;
+	return bool(*t == 0 && *p == 0);
+} // namespace duckdb
+
+// This can be moved to the scalar_function class
+template <typename Func> static void like_escape_function(DataChunk &args, ExpressionState &state, Vector &result) {
```

Comment:
Use `class OP` and `OP::Operation` for consistency with the other functions here